Tabloid/client/src/components/CategoryEditForm.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from "react";
+import { Button, Form, FormGroup, Label, Input} from "reactstrap";
+import { editCategory, getCategoryById } from "../modules/categoryManager";
+import { useHistory, useParams } from "react-router-dom";
+ 
+
+const CategoryEditForm = () => {
+  
+
+  const [category, setCategory] = useState({});
+  const history = useHistory();
+  const {id} = useParams();
+
+
+  useEffect(() => {
+    getCategoryById(id).then((category) => setCategory(category));
+  }, []);
```

Comment:
Should this useEffect listen for a change to the value of the id variable?

To Test: 
1. Click on a category to edit (example with the id of 1).
2. Manually change the URL to a different route path for a different category (example "category/edit/2")

Expected Outcome: State should be updated to reflect category 2, and the component should re-render with that category.

Reply:
When I remove the id from this use effect, an error is thrown. Currently this is working as expected.

Reply:
what I mean is, the dependency array is empty here, and it seems like the id should be observed so that this function runs every time it changes. 
```suggestion
  }, [id]);
```